plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+
+Fetching data has never been the hard part of calling APIs in React. It's always been figuring out how to cache it in a render-safe way that's been tricky.
+
+### Problems with fetching via `useAsync`
+
+While the [`useAsync` hook](https://github.com/streamich/react-use/blob/master/src/useAsync.ts) fares slightly better compared to using `useState` + `useEffect`, it still has a number of the same problems. In fact, it introduces problems.
+
+#### Problems fixed
+
+- No more race conditions
+
+#### Problems retained
+
+- Everything else from the `useEffect` section
+
+#### New problems
+
+- Even though `useAsync`'s API uses dependency arrays, by default, it is not eligible for the exhaustive deps ES Lint rule. This means that unless you update your ESLint rules, you have no safety nets for making sure that your effect logic runs the correct number of times. There are no protections against accidental typos.
+
+## Bring your own Query Client
+
+By default, `CoderProvider` manages and maintains its own `QueryClient` instance for managing all ongoing queries and mutations. This client is isolated from any other code, and especially if you are only using official Coder components, probably doesn't need to be touched.
+
+However, let's say you're using Tanstack Query for other purposes and would like all Coder requests to go through your query client instance instead. In that case, you can feed that instance into `CoderProvider`, and it will handle all requests through it instead
+
+```tsx
+<CoderProvider
+  appConfig={yourCustomAppConfig}
+  queryClient={yourCustomQueryClient}
+>
+  <YourCustomComponents />
+</CoderProvider>
+```
+
+### Warning
+
+Be sure to provide a custom query client if you put any custom components inside `CoderProvider`. Because of React Context rules, any calls to Tanstack APIs (including the `useCoderQuery` convenience wrapper) will go through the default `CoderProvider` query client, rather than your custom client.
+
+```tsx
+const customQueryClient = new QueryClient();
+
+function YourCustomComponent() {
+  const client = useQueryClient();
+  console.log(client);
+
+  return <p>Here's my content!</p>;
+}
+
+<QueryClient client={yourCustomQueryClient}>
+  {/* When mounted here, component receives customQueryClient */}
+  <YourCustomComponent />
+
+  {/*
+   * Forgot to thread customQueryClient into CoderProvider. All children will
+   * receive the default query client that CoderProvider maintains
+   */}
+  <CoderProvider>
+    {/*
+     * Exact same component definition, but the location completely changes the
+     * client that gets accessed
+     */}
+    <YourCustomComponent />
+  </CoderProvider>
+</QueryClient>;
+```
+
+## Sharing data between different queries and mutations
```

Comment:
For the same reasons, I would say we should omit this at least from this guide.  Our examples should be prescriptive and do the best thing, and if we do end up needing to document advanced use cases we could revisit.